addons/auth_oauth/models/res_users.py

```diff
@@ -10,22 +10,34 @@
 from odoo.exceptions import AccessDenied, UserError
 from odoo.addons.auth_signup.models.res_users import SignupError
 
-from odoo.addons import base
-base.models.res_users.USER_PRIVATE_FIELDS.append('oauth_access_token')
-
 
 class ResUsers(models.Model):
     _inherit = 'res.users'
 
     oauth_provider_id = fields.Many2one('auth.oauth.provider', string='OAuth Provider')
     oauth_uid = fields.Char(string='OAuth User ID', help="Oauth Provider user_id", copy=False)
-    oauth_access_token = fields.Char(string='OAuth Access Token', readonly=True, copy=False, prefetch=False)
+    oauth_access_token = fields.Char(string='OAuth Access Token Store', readonly=True, copy=False, prefetch=False, groups=fields.NO_ACCESS)
+    has_oauth_access_token = fields.Boolean(string='Has OAuth Access Token', compute='_compute_has_oauth_access_token')
 
     _uniq_users_oauth_provider_oauth_uid = models.Constraint(
         'unique(oauth_provider_id, oauth_uid)',
         'OAuth UID must be unique per provider',
     )
 
+    @api.depends('oauth_access_token')
+    @api.depends_context('uid')
+    def _compute_has_oauth_access_token(self):
+        # mask value if user has no write access on the model
+        if self.browse().has_access('write'):
+            for user in self:
+                user.has_oauth_access_token = bool(user.sudo().oauth_access_token)
+        else:
+            self.has_oauth_access_token = False
+
+    def remove_oauth_access_token(self):
+        self.check_access('write')
```

Comment:
To be honest I am a bit afraid with this `self.check_access('write')`. It's unclear to me who has the write access on res.users.

I would have allow only if `has_group('base.group_erp_manager') or self == self.env.user`